### PR TITLE
[AmazonSqsMessenger] Use AsyncAws to handle SQS communication

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | /
| License       | MIT
| Doc PR        | /

Similar to #35992 this PR use AsyncAws to handle Sqs messages sent/receive

It move complexity of authentication/streaming outside Symfony while keeping HttpClient integration.

### CODE REVIEW

**Comment by Nyholm** [comment]
If I run my application on a lambda or EC2, there is no authentication configuration required in Symfony. 

It will automatically pick up the predefined environment variables. Of course I need to configure the application's role in AWS.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (jderusse) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php]: That were a bug. When user provides an external accountId, we shouldn't try to create the queue on the Owner's account

**Comment by chalasr** [comment]
LGTM 👍 (once rebased)

**Comment by chalasr** [comment]
@jderusse Seems like the low-deps job is still failing https://travis-ci.org/github/symfony/symfony/jobs/678251415#L5486, any idea?
Needs another rebase also :)

**Comment by jderusse** [comment]
Rebased. But I think we should wait for AsyncAws version 1.0 before merging.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:243]: `resolve()` is not needed.

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:245]:
```suggestion
        // Blocking call to wait for the queue to be created
        $exists->wait();
```

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:225]:
```suggestion
        // Set to false to disable setup more than once
        $this->configuration['auto_setup'] = false;
```

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (Nyholm) [inline src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:269]: Why use `"0"` when we cast it to int?

```suggestion
        return (int) ($attributes[QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES] ?? 0);
```

**Review by Nyholm — APPROVED** [review]
Thank you. I had some minor suggestions. 

:+1:

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php
Comment (jderusse) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/AmazonSqsIntegrationTest.php:63]: thanks

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (jderusse) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:245]: indeed, comments are helpfull (applied the next suggestion too)

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (jderusse) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:243]: I don't think we should remove this one.
If user don't have permission to create the queue, the exception will be triggered on object's destruct (after calls to queueExists bellow) which can be confusing for the user

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (jderusse) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php]: thanks

File: src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php
Comment (Nyholm) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php:244]: But the Result object is destructed immediately since we don't care about the return value for `createQueue()`.

Comment (jderusse) [reply src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php]: indeed... Removed (same for `delete` method)

**Review by Nyholm — APPROVED** [review]
Im happy with the PR now

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jderusse.

**Pull request merged**
